Peek-a-boo header strip

bradchoate opened the pull request:
## Overview

Implementation of 'peek-a-boo' site header.

## Screenshots

![image](https://github.com/user-attachments/assets/d60e4301-ea10-4728-b06e-dd621ffca8c3)

## Testing

Available in Storybook under [Legacy -> Components -> Site Header -> Logged in With Content](https://deploy-preview-1317--mltshp-patterns.netlify.app/?path=/story/legacy-components-site-header--logged-in-with-content)

Scroll down until header area is off-screen, then scroll up to reveal the header. Scrolling downward will hide it again. There is no non-interaction timer for automatic hiding after reveal.

---

- Fixes #1302

spaceninja commented:
Code looks fine. The design is a bit clunky, but it always was going to be, switching between the proper header and the peekaboo header. I like that you didn't have to duplicate the content for the peekaboo nav.

The one question I have is about on small screens — I see that the buttons go away and are replaced with the menu button, just like the proper header. Will the menu still be positioned properly if you're scrolled away from the top when it opens? I assume it'll work, since I'm guessing it's positioned relative to the header, not the page, but I just wanted to confirm that.

That's not a blocker, though, since this is just the pattern library representation, so I guess I'm just making a mental note to check that when we implement it on the actual site.

I'll defer on design approval to @heyitsal, but approving now for code review.

spaceninja commented:
Oh I guess I did have one question — Should the peekaboo menu's backdrop be white for light mode? It looks great in dark mode, but it's a little jarring in light mode.

Here's an example of white:
<img width="1091" alt="Screenshot 2025-02-23 at 12 52 38 PM" src="https://github.com/user-attachments/assets/a0cc21d9-8c15-4654-af57-e222b09bda0d" />

And here's one with the light blue we use for the page background:
<img width="1043" alt="Screenshot 2025-02-23 at 12 54 46 PM" src="https://github.com/user-attachments/assets/ee3882bc-28fa-473e-b164-aea450ac7804" />

I think both look a little more natural for light mode, but I'll defer to Al.

jessamynwest commented:
I'm definitely a small screen person who would be happy to give this a look-see when it's on the live site.

bradchoate commented:
> I'm definitely a small screen person who would be happy to give this a look-see when it's on the live site.

You can play with it here: https://deploy-preview-1317--mltshp-patterns.netlify.app/?path=/story/legacy-components-site-header--logged-in-with-content

bradchoate commented:
Ah I forgot to check light mode. I can address that easily enough. The slide in/out should also occur once scrolling starts not once it ends; I'll address that as well. Finally, I think we could do something with the mobile version since there is a lot of empty space left after condensing the logo to an icon. Any thoughts on that?

bradchoate commented:
> Will the menu still be positioned properly if you're scrolled away from the top when it opens? I assume it'll work, since I'm guessing it's positioned relative to the header, not the page, but I just wanted to confirm that.

Looks like it just works (manually applied the `is-expanded` class to `#site-nav` and `#choose-a-shake`.

![image](https://github.com/user-attachments/assets/7a5403ba-e4fe-44da-8a31-4505228e4a0e)

Also updated the background for light mode, and removed the debounced function so it shows/hides the nav strip more promptly on scroll.

bradchoate commented:
Special consideration may be needed for the expanded menus... if that menu is very long and requires scrolling, scrolling downward may cause the header and menus to disappear as it is (although, the dismiss on scroll does not occur for gradual scrolling... you have to scroll > 20px since last scroll event to trigger it).

bradchoate commented:
> Special consideration may be needed for the expanded menus... if that menu is very long and requires scrolling, scrolling downward may cause the header and menus to disappear as it is (although, the dismiss on scroll does not occur for gradual scrolling... you have to scroll > 20px since last scroll event to trigger it).

Updated the PR to restrict maximum height of the shake list to `80vh` and scroll on overflow. This resolves the case where there are too many shakes to be contained in a single viewport.

bradchoate commented:
> The design is a bit clunky, but it always was going to be

Can you elaborate with specific details? Anything I can improve on?

spaceninja commented:
> Can you elaborate with specific details?

Honestly, upon further reflection, I think it was mostly that the dark background looked weird in light mode.

I clicked through to the Storybook preview deploy to review your latest changes, but the peekaboo isn't working? I'm using Edge.

bradchoate commented:
> > Can you elaborate with specific details?
> 
> Honestly, upon further reflection, I think it was mostly that the dark background looked weird in light mode.
> 
> I clicked through to the Storybook preview deploy to review your latest changes, but the peekaboo isn't working? I'm using Edge.

Working for me in Edge (latest). Any console errors?

spaceninja commented:
> Working for me in Edge (latest). Any console errors?

After testing again, I now see the peekaboo header with the same cut-off bug you described above. Sorry!

bradchoate commented:
Any ideas on how to better use the space for mobile? It could be used to show notifications. Tapping that # could scroll to the anchor (at the bottom of the page). (Spacing issues as pictured, but would clean that up.)

![image](https://github.com/user-attachments/assets/08d2e884-fbf9-43bf-9ac6-28fe47f54d39)

![image](https://github.com/user-attachments/assets/fc944a79-be35-4b43-8b5f-2cc8fb966004)

bradchoate commented:
Say, regarding the cut-off thing... I sometimes see this happen on image content on the site. I've only ever seen it happen on Chromium-based browsers (I'm using Arc these days, but pretty sure it's happened in Chrome too). Any idea what could be causing it? It's an odd rendering glitch. If I drag to select the image causing it to repaint, it does.

![image](https://github.com/user-attachments/assets/e9e4968e-e2bb-48ea-b45d-42439b40ede7)
